src/MBX/fix_MBX.h

```diff
@@ -0,0 +1,220 @@
+
+  MBXT_DISP_PME_SETUP,
+  MBXT_DISP_PME_E,
+
+  MBXT_NUM_TIMERS
+};
+
+struct MBXParseResult {
+  bool success;
+  std::string message;
+};
+
+namespace LAMMPS_NS {
+
+class FixMBX : public Fix {
+  friend class PairMBX;
+
+ public:
+  FixMBX(class LAMMPS *, int, char **);
+  ~FixMBX();
+  int setmask();
+  virtual void post_constructor();
+  virtual void init();
+  virtual void init_storage();
+  void setup(int);
+  void min_setup(int);
+  virtual void setup_post_neighbor();
+  virtual void post_neighbor();
+  void min_post_neighbor();
+  void setup_pre_force(int);
+  virtual void pre_force(int);
+
+  void min_setup_pre_force(int);
+  void min_pre_force(int);
+
+  void setup_pre_exchange();
+  void pre_exchange();
+
+  void post_force(int);
+  void min_post_force(int);
+
+ protected:
+  class PairMBX *pair_mbx;    // pointer to MBX pair_style
+
+  bblock::System *ptr_mbx;          // pointer to MBX object
+  bblock::System *ptr_mbx_full;     // pointer to MBX object for full system
+  bblock::System *ptr_mbx_local;    // pointer to MBX object for local atoms
+
+  int me, nprocs;
+  bigint ngroup;
+
+  bool mbx_mpi_enabled;
+  bool mbx_aspc_enabled;
+  bool print_dipoles;
+
+  bool first_step;
+  bool has_gcmc;
+
+  int use_json;
+  char *json_file;
+  std::string json_settings;
+
+  int print_settings;
+
+  int num_mol_types;    // # of unique molecule types
+  int num_molecules;    // total # of molecules
+  // int *num_mols;           // array of # of molecules of each type
+  int *num_atoms_per_mol;                // array of # of atoms per molecule for each type
+  int *lower_atom_type_index_in_mol;     // array with the lowest atom type index in the monomer
+  int *higher_atom_type_index_in_mol;    // array with the highest atom type index in the monomer
+  int **order_in_mol;                    // array with the atom order for each monomer
+  // tagint *mol_offset;      // array of atom count per molecule prefix sum
+  char **mol_names;    // array of molecule names
+
+  int *mol_type;      // per-atom array of molecule type
+  int *mol_anchor;    // per-atom array 1/0 if anchor atom of a molecule
+  // int *mol_order;   // per-atom array 1/2/3/4... with position of atom in molecule
+  int *mol_local;    // per-molecule array 1/0 if molecule has at least one local particle
+
+  int mbx_num_atoms, mbx_num_ext;
+  int mbx_num_atoms_full, mbx_num_ext_full;
+  int mbx_num_atoms_local, mbx_num_ext_local;
+
+  int *mbxt_count;
+  double *mbxt_time;
+  double *mbxt_time_start;
+  double mbxt_initial_time;
+  void mbxt_start(int);
+  void mbxt_stop(int);
+  void mbxt_write_summary();
+  void mbxt_print_time(const char *, int, double *);
+
+  bool validateMBXFixParameters(int narg, char **arg);
+
+  int aspc_order;
+  int aspc_num_hist;
+  int aspc_max_num_hist;
+  int aspc_per_atom_size;
+  double **aspc_dip_hist;
+
+  int aspc_step;
+  int aspc_step_reset;
+
+  double **mbx_dip;
+
+  // rank 0's copy of all atoms in simulation cell
```

Comment:
Is this comment true?? Does the fix have to hold a copy of the entire system? This would be a *very big* problem as this seriously limits the use of this pair style inside LAMMPS and invalidates all the effort in LAMMPS to distribute the per-atom data across subdomains. I would have to talk to the other LAMMPS developers, if we want to include a package with such a limitation.